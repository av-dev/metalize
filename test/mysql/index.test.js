'use strict';

const helpers = require('../helpers');

const options = {
  dialect: 'mysql',
  connectionConfig: {
    host: 'localhost',
    user: 'root',
    database: 'mysql',
    port: 3306,
  },
};

describe(`'${options.dialect}' dialect`, () => {
  helpers.setup({
    ...options,
    schema: options.connectionConfig.database,
    onGotAdditionalBlocks: metalize => {
      it('unsupported sequence reading', async () => {
        try {
          await metalize.read.sequences(['sequence_name']);
        } catch (e) {
          return true;
        }
        throw new Error('error test');
      });
    },
  });
});
