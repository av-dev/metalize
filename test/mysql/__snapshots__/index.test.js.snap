// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'mysql' dialect table metadata 1`] = `
Result {
  "sequences": Map {},
  "tables": Map {
    "metalize_schema.undefined_table" => undefined,
    "metalize_schema.users" => Object {
      "checks": Array [],
      "columns": Array [
        Object {
          "default": null,
          "identity": false,
          "name": "id",
          "nullable": false,
          "type": "bigint",
        },
        Object {
          "default": "noname",
          "identity": false,
          "name": "name",
          "nullable": true,
          "type": "varchar(255)",
        },
        Object {
          "default": null,
          "identity": false,
          "name": "budget",
          "nullable": true,
          "type": "decimal(16,3)",
        },
        Object {
          "default": null,
          "identity": true,
          "name": "age",
          "nullable": false,
          "type": "bigint",
        },
        Object {
          "default": null,
          "identity": false,
          "name": "child",
          "nullable": true,
          "type": "bigint",
        },
      ],
      "foreignKeys": Array [
        Object {
          "columns": Array [
            "id",
            "child",
          ],
          "match": "SIMPLE",
          "name": "users_f_constraint",
          "onDelete": "CASCADE",
          "onUpdate": "RESTRICT",
          "references": Object {
            "columns": Array [
              "parent",
              "id",
            ],
            "table": "metalize_schema.users_child",
          },
        },
      ],
      "indexes": Array [
        Object {
          "columns": Array [
            "age",
          ],
          "name": "age_index",
        },
        Object {
          "columns": Array [
            "id",
            "child",
          ],
          "name": "index_name",
        },
      ],
      "name": "metalize_schema.users",
      "primaryKey": Object {
        "columns": Array [
          "id",
        ],
        "name": "PRIMARY",
      },
      "unique": Array [
        Object {
          "columns": Array [
            "name",
            "age",
          ],
          "name": "users_u_constraint",
        },
      ],
    },
  },
}
`;
